### ISSUE TITLE
[Feature Request] Add new badges for friended/followed accounts

### ISSUE THREAD

**Issue by synap5e** [FrankerFaceZ/FrankerFaceZ#760]
**Is your feature request related to a problem? Please describe.**
- Some streamers I follow have communities or related streamers where they frequent each other's chats. The result is that the "Broadcaster" badge is not the only user in chat that is of specific interest. 
- Additionally it can be interesting to see streamers I follow in large chats e.g. tournament streams where they would be otherwise easy to miss. This is especially the case if they are streaming their participation in the tournament and chime in on the main stream.
- Finally, being able to easily notice when a friend is chatting in a stream that I am not paying attention to the chat of would be nice.

**Describe the solution you'd like**
- An option to add special badges to followed accounts and friended accounts.

**Describe alternatives you've considered**
- Manually adding the accounts I follow and my friends to the highlighted users list. This is less ideal since it requires constantly manually updating the list, and highlights are less ideal than badges since:
  - Badges are more suited for adding contextual information regarding a user which is what this would be aiming to do.
  - Highlights are a much less "fine-grained" solution, and far harder to tune out. Badges are easy to spot if you want to check for them, but easy to ignore if you are not interested at the time.

**Additional context**
This issue is similar to #116 #589 (which have been merged into #299) but proposes to use badges instead of highlights for the above reasons.

https://github.com/FrankerFaceZ/FrankerFaceZ/issues/598#issuecomment-495727310 raises the issue regarding using the list of followed streamers
> This isn't really viable as the client never requests a full list of your followed streamers during normal operations, and it isn't something I would want to trigger myself.

To solve this issue, perhaps this feature could wait on #141 (#462 #132) since completing that could add a mechanism for fetching all followed accounts.

**Comment by SirStendec** [comment]
As you pointed out, I have rejected a very similar issue as not viable because the client never requests a full list of your followed streamers.

For the exact same reason, I reject this. Whether a badge or a line highlight, they're not fundamentally different. You'd still need to load a potentially massive list into the client. Expanding the list of the sidebar by a few entries has nothing to do with loading a full list of followers. It'd be like... 50 entries max when you can follow hundreds of channels.

**Comment by synap5e** [comment]
My thinking was populating the sidebar would require loading at least one results page of followed channels which could fetch up to 100 channels with the results used for both the sidebar and this feature. Pagination would be the issue, since loading the sidebar would only care about the first page. Personally only supporting the first 100 followed users would work for me and my gut instinct is this would cover most users, although I understand if you don't want to feature with such a caveat.

I do request you at least consider using badges for friends instead of highlighting, if this is still on the roadmap. I feel badges are a lot less obtrusive, although I realize we could have diverging ideas on what badges are for. As far as I can tell though, friends support would also require querying `OnlineFriends` and suffer similar pagination issues. Being gql adding a `FollowedChannels` to the same query should be easy and cheap.

The badge system already loads close to 40k entries for supporters across 2 api calls (and some more for bttv?). Looking at the timings, this is taking ~2.5s in total, while complex GQL queries are completing in 100s of ms so even with pagination I don't believe this would be a massive problem especially if rolled into the same friend fetching gql call.
